workflows/iuf/operations/extract-release-distributions.yaml

```diff
@@ -26,38 +26,80 @@ kind: WorkflowTemplate
 metadata:
   name: extract-release-distributions
 spec:
+  tolerations:
+    - key: "node-role.kubernetes.io/master"
+      operator: "Exists"
+      effect: "NoSchedule"
+  nodeSelector:
+    kubernetes.io/hostname: ncn-m001
   entrypoint: main
   templates:
     - name: main
       inputs:
         parameters:
           - name: auth_token
           - name: global_params
-      retryStrategy:
-        limit: "2"
-        retryPolicy: "Always"
-        backoff:
-          duration: "10s" # Must be a string. Default unit is seconds. Could also be a Duration, e.g.: "2m", "6h", "1d"
-          factor: "2"
-          maxDuration: "1m"
+      dag:
+        tasks:
+          - name: list-tar-files
+            inline:
+              script:
+                image: artifactory.algol60.net/csm-docker/stable/iuf:v0.0.4
+                command: [sh]
+                source: |
+                  #!/bin/sh
+                  set -x
+                  media_dir="{{=jsonpath(inputs.parameters.global_params, '$.input_params.media_dir')}}"
+                  cd $media_dir && ls *.tar.gz | jq -R -s -c 'split("\n")[:-1]'
+                volumeMounts:
+                  - name: iuf
+                    mountPath: /opt/cray/iuf
+              volumes:
+                - name: iuf
+                  hostPath:
+                    path: /opt/cray/iuf
+                    type: Directory
+          - name: extract-tar-files
+            dependencies:
+              - list-tar-files
+            arguments:
+              parameters:
+                - name: tarfile
+                  value: "{{item}}"
+                - name: media_dir
+                  value: "{{=jsonpath(inputs.parameters.global_params, '$.input_params.media_dir')}}"
+            withParam: "{{tasks.list-tar-files.outputs.result}}"
+            template: extract-tar-files
+    - name: extract-tar-files
+      inputs:
+        parameters:
+          - name: tarfile
+          - name: media_dir
       script:
         image: artifactory.algol60.net/csm-docker/stable/docker.io/alpine/git:2.32.0
         command: [sh]
         source: |
           #!/bin/sh
-          tarfile="{{=jsonpath(inputs.parameters.global_params, '$.product_manifest.current_product.original_location')}}"
-          productName="{{=jsonpath(inputs.parameters.global_params, '$.product_manifest.current_product.name')}}"
-          echo "Extract from: $tarfile for product: $productName"
-          dir=$(dirname "$tarfile")
-          parent_directory=${tarfile%.tar.*}
-          echo ${parent_directory} > /tmp/parent_directory
-          cd $dir
-          tar -xvzf $tarfile
+          set -x
+          media_dir={{inputs.parameters.media_dir}}
+          cd $media_dir
+          tar -xvzf {{inputs.parameters.tarfile}} &> /dev/stderr
+          dir_name=`tar -tzf {{inputs.parameters.tarfile}} | head -1 | cut -f1 -d"/"`
+          cat ${dir_name}/iuf-product-manifest.yaml > /tmp/manifest.yaml
```

Comment:
This doesn't look like it will fail gracefully if any tar file contained in this directory does not contain an IUF product manifest. I think we want to handle that somehow, don't we?

Reply:
the operator is dealing with that, plus validation

Reply:
My misunderstanding was based on the fact that I thought this was checking the exit code of the `cat` command. It is not. If the manifest file isn't there, the `/tmp/manifest.yaml` file is still created, but it's empty, and the output parameter is empty. The IUF operator handles that somewhat gracefully here: https://github.com/Cray-HPE/cray-nls/blob/fe275b4dc5a068446a76b360f397683d044a39d0/src/api/services/iuf/sessions.go#L376

This results in kind of a strange error, but it does not break anything. E.g.:

```
2022-11-30T21:28:25.788Z        INFO    iuf/sessions.go:376     manifest: %!s(<nil>) - %!s(<nil>)
```